### DISCUSSION TITLE
Deterministic tests

### DISCUSSION THREAD

**Discussion by valeriob01** [preda/gpuowl#212] [Ideas]
Instead of using a probabilistic algorithms like PRP, why don't use a deterministic one like https://en.wikipedia.org/wiki/Adleman%E2%80%93Pomerance%E2%80%93Rumely_primality_test, in particular this implementation https://sourceforge.net/projects/mpzaprcl/ ?

**Comment by valeriob01** [comment]
> Instead of using a probabilistic algorithms like PRP, why don't use a deterministic one like https://en.wikipedia.org/wiki/Adleman%E2%80%93Pomerance%E2%80%93Rumely_primality_test, in particular this implementation https://sourceforge.net/projects/mpzaprcl/ ?

Do you find this question off topic? or maybe not meriting reply?

**Comment by preda** [comment]
I already replied in the "AKS" topic: these general primality tests are much slower than the mersenne-specific tests that we use (LL and PRP). It's not clear what "much slower" is, but I expect, for 2^100M numbers, to be on the order of 1000x times slower.

**Reply by valeriob01** [reply]
> I already replied in the "AKS" topic: these general primality tests are much slower than the mersenne-specific tests that we use (LL and PRP). It's not clear what "much slower" is, but I expect, for 2^100M numbers, to be on the order of 1000x times slower.

I'm not a math-guru like you (as I would like to be) thus I ask probably useless questions :-)
